src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (health > 0) {
+      Animal.alive.push(this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (animal.hidden !== true && animal instanceof Herbivore) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        const index = Animal.alive.indexOf(animal);
+
+        if (index !== -1) {
+          Animal.alive.splice(index, 1);
+        }
+      }
```

Comment:
it would be nice, if you would `console.error` when e.g. we want to bite hidden animal. 
to make it easier write your `if statements` with [guard clauses](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/) 

so that you can check
```
if(!something) {
//log error and return
}
```